src/contracts/data-bus/data-bus.client.ts

```diff
@@ -0,0 +1,147 @@
+import { Contract, providers, Signer, utils } from 'ethers';
+import { EventDataMap, eventMappers } from './data-bus.serializer';
+import { MessagesDataMap, MessagesNames } from './data-bus.serializer';
+import * as eventsAbi from '../../abi/data-bus.abi.json';
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { DATA_BUS_REQUEST_TIMEOUT } from './data-bus.constants';
+
+export class DataBusClient {
+  private dataBusAddress: string;
+  private eventsInterface: utils.Interface;
+  private provider: providers.Provider;
+  private eventsFragments: utils.EventFragment[] = [];
+  private dataBus: Contract;
+
+  constructor(dataBusAddress: string, signer: Signer) {
+    this.dataBusAddress = dataBusAddress;
+    this.eventsInterface = new utils.Interface(eventsAbi);
+
+    if (!signer.provider) {
+      throw new Error('Signer with provider is required');
+    }
+    this.provider = signer.provider;
+    this.eventsFragments = Object.values(this.eventsInterface.events);
+    this.dataBus = new Contract(
+      dataBusAddress,
+      ['function sendMessage(bytes32 _eventId, bytes _data)'],
+      signer,
+    );
+  }
+
+  async sendTransaction(eventId: string, dataBytes: string) {
+    const tx: TransactionResponse = await this.dataBus.sendMessage(
+      eventId,
+      dataBytes,
+    );
+    await tx.wait();
+    return tx;
+  }
```

Comment:
Should we manually increase priority fee (up to certain limit relative to current base fee) during high network activity?

Reply:
Also, do we need a retry mechanism?

Reply:
Should we resend lost messages after a chain reorg?

Reply:
answered in the chat